src/wasm/wat-parser-internal.h

```diff
@@ -68,6 +68,9 @@ struct LexCtx {
   // The next input that has not already been lexed.
   std::string_view next() const { return input.substr(lexedSize); }
 
+  // Get the next character without consuming it.
+  uint8_t peek() const { return next()[0]; }
```

Comment:
I think this can even take an optional integer argument for length (which defaults to 1)

Reply:
So far I haven't ever needed to peek beyond the next character, but if a use case for that comes up, I will definitely update the method.